spi/src/main/java/org/opensearch/security/spi/resources/Resource.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources;
+
+import java.io.IOException;
+
+import org.opensearch.core.common.io.stream.NamedWriteable;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.xcontent.ToXContentFragment;
+
+/**
+ * Marker interface for all resources
+ */
+public abstract class Resource implements NamedWriteable, ToXContentFragment {
```

Comment:
Why is this a `NamedWriteable` and not just a `Writeable`?

Reply:
I anticipate the need for using unique names for resource identification down the line.